src/FSharpPlus/Data/NonEmptyList.fs

```diff
@@ -39,6 +39,11 @@ module NonEmptyList =
     let singleton x = {Head = x; Tail = []}
     let toList {Head = x; Tail = xs} = x::xs
     let toSeq  {Head = x; Tail = xs} = seq { yield x; yield! xs; }
+    let toArray nel = toList nel |> List.toArray
+    let ofArray (array : _ array) = match Array.tryHead array with | Some head -> create head (Array.tail array |> Array.toList) | _ -> failwith "cannot construct empty non empty list"
+    let ofList (list : _ list) = match List.tryHead list with | Some head -> create head (List.tail list) | _ -> failwith "cannot construct empty non empty list"
```

Comment:
Can't we remove `Unchecked.toNonEmptyList` now?

Reply:
Sure :smil